### ISSUE TITLE
[apex] ApexCRUDViolation - DML Inside of Apex Properties Not Evaluated for User_Mode AccessLevel

### ISSUE THREAD

**Issue by stephen-carter-at-sf** [pmd/pmd#5463]
**Affects PMD Version:** 7.9.0

**Rule:**

ApexCRUDViolation
https://docs.pmd-code.org/pmd-doc-7.9.0/pmd_rules_apex_security.html#apexcrudviolation      

**Description:**

SELECT queries / DML inside of Apex properties missing AccessLevel.USER_MODE are not flagged for ApexCRUDViolation.

**Code Sample demonstrating the issue:**

```java
public with sharing class MyClass{
   private ID m_pid;
   
   public MyClass(ID parentId){
      this.m_pid = parentId; 
   }
   
   public List<SObject> ChildRecords{
      get{
        // Missing WITH USER_MODE. Not flagged for ApexCRUDViolation.
         return [SELECT Id, Name FROM ChildRecord__c WHERE ParentId=:m_pid];
      }
   }
   
   public List<SObject> queryChildRecords()
         // Missing WITH USER_MODE. Flagged for ApexCRUDViolation.
         return [SELECT Id, Name FROM ChildRecord__c WHERE ParentId=:m_pid]; 
   }
}
```

**Expected outcome:**

PMD should report a violation at line 11, but doesn't. This is a false-negative.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

CLI

But originally this issue was reported via the Salesforce Code Analyzer:
https://github.com/forcedotcom/sfdx-scanner/issues/1716
